README.md

```diff
@@ -1,3 +1,162 @@
-# Ansible Collection - firstset.fogo_community
+# FOGO Community Ansible Collection
 
-The collection for FOGO validators to maintain their node in an easier way.
+[![Ansible Galaxy](https://img.shields.io/badge/ansible--galaxy-firstset.fogo__community-blue.svg)](https://galaxy.ansible.com/ui/repo/published/firstset/fogo_community/)
+[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
+
+An Ansible collection that provides automated deployment and management tools for FOGO validators. This collection simplifies the process of setting up, configuring, and maintaining FOGO validator nodes.
+
+## Overview
+
+FOGO is a Solana validator client built on Firedancer, designed for high performance and reliability. This Ansible collection automates the deployment process, making it easier for validators to maintain their nodes with best practices and consistent configurations.
+
+## Features
+
+- **Automated validator deployment** - Complete setup from system configuration to service deployment
+- **Performance optimization** - CPU performance mode configuration and system tuning
+- **Security hardening** - UFW firewall configuration and service user management
+- **Disk management** - Automatic detection and formatting of additional storage disks
+- **Binary management** - Automated building and upgrading of Firedancer binaries
+- **Configuration templating** - Flexible Firedancer configuration with sensible defaults
+- **Service management** - Systemd service creation and lifecycle management
+
+## Requirements
+
+- **Ansible**: >= 2.9
+- **Target OS**: Ubuntu 20.04+ or Debian 11+
+- **Hardware**: Meets [FOGO's official requirements](https://docs.fogo.io/running-a-node.html)
+- **Network**: Open ports for gossip (8001) and dynamic port range (8901-9000)
+
+## Installation
+
+Install from Ansible Galaxy:
+
+```bash
+ansible-galaxy collection install firstset.fogo_community
+```
+
+Or install from source:
+
+```bash
+git clone https://github.com/Firstset/fogo-community-ansible.git
+cd fogo-community-ansible
+ansible-galaxy collection build
+ansible-galaxy collection install firstset-fogo_community-*.tar.gz
+```
+
+## Quick Start
+
+### Basic Validator Setup
+
+Create a playbook to deploy a FOGO validator:
+
+```yaml
+---
+- name: Deploy FOGO Validator
+  hosts: validator_nodes
+  become: true
+  roles:
+    - firstset.fogo_community.validator_service
+```
+
+Run the playbook:
+
+```bash
+ansible-playbook -i inventory site.yml --ask-become-pass
+```
+
+### Advanced Configuration
+
+Override default settings by setting variables:
+
+```yaml
+---
+- name: Deploy FOGO Validator with Custom Config
+  hosts: validator_nodes
+  become: true
+  vars:
+    validator_client_version: v6.0.0
```

Comment:
```suggestion
    validator_client_version: v6.0.0
    validator_client_tarfile_checksum: 817533105183734d5f4dffb4f0b11b0de2adf38b
```

When downloading a tarfile, currently the role always verifies the SHA1 checksum provided by FOGO's [release page](https://docs.fogo.io/releases.html). So I would suggest adding `validator_client_tarfile_checksum` here as well.